plugins/catalog/src/components/RelatedEntitiesCard/RelatedEntitiesCard.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import {
+  CodeSnippet,
+  InfoCard,
+  Link,
+  Progress,
+  TableColumn,
+  WarningPanel,
+} from '@backstage/core';
+import { Typography } from '@material-ui/core';
+import {
+  EntityTable,
+  useEntity,
+  useRelatedEntities,
+} from '@backstage/plugin-catalog-react';
+import React from 'react';
+
+type Props<T extends Entity> = {
+  variant?: 'gridItem';
+  title: string;
+  columns: TableColumn<T>[];
+  entityKind: string;
+  relationType: string;
+  emptyMessage: string;
+  emptyHelpLink: string;
+  asRenderableEntities: (entities: Entity[]) => T[];
+};
+
+export function RelatedEntitiesCard<T extends Entity>({
+  variant = 'gridItem',
+  title,
+  columns,
+  entityKind,
+  relationType,
+  emptyMessage,
+  emptyHelpLink,
+  asRenderableEntities,
+}: Props<T>) {
+  const { entity } = useEntity();
+  const { entities, loading, error } = useRelatedEntities(entity, {
+    type: relationType,
+    kind: entityKind,
+  });
+
+  if (loading) {
+    return (
+      <InfoCard variant={variant} title={title}>
+        <Progress />
+      </InfoCard>
+    );
+  }
+
+  if (error || !entities) {
+    return (
+      <InfoCard variant={variant} title={title}>
+        <WarningPanel
+          severity="error"
+          title={`Could not load ${{ title }}`}
+          message={<CodeSnippet text={`${error}`} language="text" />}
+        />
+      </InfoCard>
+    );
+  }
+
+  return (
+    <EntityTable
```

Comment:
Whoa I just now remembered / realized that the table actually just sort of mimics the look of a card. And the paddings and stuff are different.

If it didn't, it would have been neater to render the InfoCard in the outer component (e.g. `HasComponentsCard`), and make this inner component `RelatedEntitiesTable` with fewer props.

OK, something to look into another day. Doesn't affect this review :)